Functions/xASL_Iteration.m

```diff
@@ -41,65 +48,80 @@
+        if ~isfield(x.settings,'stopAfterErrors') || isempty(x.settings.stopAfterErrors)
+            dbSettings.settings.stopAfterErrors = Inf;
+        else
+            dbSettings.settings.stopAfterErrors = x.settings.stopAfterErrors;
+        end
+    else
+        dbSettings.settings.stopAfterErrors = stopAfterErrors;
     end
-	    
+    
+    % Check if it was possible to load the subjects
     if nargin<5 || isempty(SelectedSubjects)
-        try
+        if isfield(x,'SUBJECTS')
             SelectedSubjects = x.SUBJECTS;
-        catch ME
-            warning('Loading subjects didnt work, verify that ExploreASL was correctly initialized with the correct DataParameter file');
-            fprintf('%s\n', ME.message);
+        else
+            warning('Loading subjects didnt work, verify that ExploreASL was correctly initialized with the correct DataParameter file...');
         end
     end
     
-    dbSettings.jobfn                 = str2func(moduleName);
-    fprintf(['Running ' moduleName ' ...   ']);
-
+    % Print module name
+    dbSettings.jobfn = str2func(moduleName);
+    fprintf('\nRunning %s...\n', moduleName);
+    
+    % Get module name string
     if  strcmp(moduleName(1:12),'xASL_module_')
-        ModName     = moduleName(13:end);
-    elseif strcmp(moduleName(1:7),'module_') % backward compatibility
-        ModName     = moduleName(8:end);
+        ModName = moduleName(13:end);
+    elseif strcmp(moduleName(1:7),'module_')
+        % backward compatibility
+        ModName = moduleName(8:end);
     else
-        ModName     = moduleName;
+        ModName = moduleName;
     end
-
+    
     % Specific settings
     if  strcmp(ModName,'DARTEL') || strcmp(ModName,'LongReg')
-        dbSettings.x.dir.LockDir         = [dbSettings.x.dir.LockDir '_' x.P.STRUCT ];
-	end
+        dbSettings.x.dir.LockDir = [dbSettings.x.dir.LockDir '_' x.P.STRUCT ];
+    end
     
-	if ~isempty(regexp(ModName,'(Struct|ASL|func|LongReg|dwi)'))
-		dbSettings.sets.SUBJECT      = SelectedSubjects; % x.SUBJECTS
-		dbSettings.x.dir.SUBJECTDIR  = '<ROOT>/<SUBJECT>';
-		dbSettings.x.dir.LockDir     = [dbSettings.x.dir.LockDir '/<SUBJECT>'];
-	end
-	
-	if ~isempty(regexp(ModName,'(ASL|func|dwi)'))
-		dbSettings.sets.SESSION       = x.SESSIONS;
-		dbSettings.x.settings.MUTEXID = [dbSettings.x.settings.MUTEXID '_<SESSION>'];
-		dbSettings.x.dir.SESSIONDIR       = '<ROOT>/<SUBJECT>/<SESSION>';
-	end
-	
-	if ~isempty(regexp(ModName, '(DARTEL|Population|Analyze)'))
-		dbSettings.diaryFile               = ['<PopDir>/' moduleName '.log'];
-	elseif ~isempty(regexp(ModName,'(ASL|func|dwi)'))
-		dbSettings.diaryFile               = ['<SESSIONDIR>/' moduleName '.log'];
-	else
-		dbSettings.diaryFile               = ['<SUBJECTDIR>/' moduleName '.log'];
-	end
-	
-	[bAborted, xOut] = runIteration(dbSettings);
+    % SUBJECT, SUBJECTDIR & LockDir
+    if ~isempty(regexp(ModName,'(Import|Struct|ASL|func|LongReg|dwi)', 'once'))
```

Comment:
yes, this would work. No reason this wouldn't work. It's the combination with `isempty` that's important. It doesn't matter if it is "not-empty" once or twice ;)

Reply:
Exactly like @jan-petr said 👍